src/server/api.php

```diff
@@ -991,12 +992,41 @@ function get_covidcast($source, $signals, $time_type, $geo_type, $time_values, $
   } else {
     // fetch most recent issue fast
     $condition_version = '(t.`is_latest_issue` IS TRUE)';
+    $union = False;
   }
   // the query
-  $query = "SELECT {$fields} FROM {$table} {$subquery} WHERE {$conditions} AND ({$condition_version}) ORDER BY {$order}";
+  $query_template = "SELECT {$fields} FROM {$table} {$subquery} WHERE {$conditions} AND ({$condition_version})";
+
+  $from_2021 = str_replace("#TABLE#", "covidcast", $query_template);
+  $query = "";
+  if ($union) {
+    $from_2020 = str_replace("#TABLE#", "covidcast2", $query_template);
+    $query = "({$from_2020}) UNION ({$from_2021})";
+    $order = "{$order}, `value_updated_timestamp` DESC";
+  } else {
+    $query = $from_2021;
+  }
+
+  $query = "{$query} ORDER BY {$order}";
   // get the data from the database
   $epidata = array();
   execute_query($query, $epidata, $fields_string, $fields_int, $fields_float);
+
+  // drop duplicates where some source+signal+geo+date is in both tables with
+  // different timestamps, preferring most recently updated copy
+  if ($union) {
+    $drop = array();
```

Comment:
can you add a comment that this logic requires that the rows are sorted such that the same keys are after each other and that the `value_updated_timestamp` is deceasing.

At the moment it is not clear the rder of the SQL statement and this logic is heavily dependent on each other